scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,358 @@
+    in_mask             Path to binary brainmask
+    lower_threshold     Value for low thresold <int>
+    upper_thresold      Value for up thresold <int>
+    idx_contrast_list   List of indice contrasts maps in order corresponding to
+                        contrast_maps input ex.: [1, 2, 3]
+                        Altnp = 0; Atlpn = 1; Reference = 2; Negative = 3;
+                        Positive = 4; T1weighted = 5
+    Returns
+    ----------
+    Thresholded map     3D-array
+    """
+    # Remove NaN and apply thresold based on lower and upper value
+    computed_map[np.isnan(computed_map)] = 0
+    computed_map[np.isinf(computed_map)] = 0
+    computed_map[computed_map < lower_threshold] = 0
+    computed_map[computed_map > upper_threshold] = 0
+
+    # Load and apply brainmasks on maps
+    mask_image = nib.load(in_mask)
+    mask_data = mask_image.get_fdata(dtype=np.float32)
+    computed_map[np.where(mask_data == 0)] = 0
+
+    # Apply threshold based on combination of specific contrasts maps
+    for idx in idx_contrast_list:
+        computed_map[contrast_maps[idx] == 0] = 0
+
+    return computed_map
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    # Select contrasts files from Bids folder
+    maps = []
+    layout = BIDSLayout(args.in_bids_folder, validate=False)
+    acquisition = layout.get_acquisition()
+
+    if args.resolution:
+        curr_res_acq = [res for res in acquisition if args.resolution[0] in res]
+        for curr_acq in curr_res_acq:
+            maps.append(layout.get(subject=args.in_participant,
+                                   datatype='anat', suffix='ihmt',
+                                   acquisition=str(curr_acq),
+                                   extension='nii.gz', return_type='file'))
+    else:
+        for curr_acq in acquisition:
+            maps.append(layout.get(subject=args.in_participant, datatype='anat',
+                                   suffix='ihmt', acquisition=str(curr_acq),
+                                   extension='nii.gz', return_type='file'))
+
+    # Set TR and FlipAngle parameters for ihMT and T1w images
+    parameters = []
+    parameters.append(set_acq_parameters(layout, maps[4][0]))
+    parameters.append(set_acq_parameters(layout, maps[5][0]))
+
+    # Fix issue from the presence of NaN into array
+    np.seterr(divide='ignore', invalid='ignore')
+
+    # Define reference image for saving ihMT files
+    ref_img = nib.load(maps[4][0])
+
+    # Create contrasts maps folers
+    if not os.path.isdir(os.path.join(args.out_dir, 'Contrats_maps')):
+        os.mkdir(os.path.join(args.out_dir, 'Contrats_maps'))
+
+    # Compute contrasts maps
+    computed_contrasts = []
+    outname = ['altnp', 'altpn', 'reference', 'negative', 'positive', 'T1w']
+    for idx, curr_map in enumerate(maps):
+        if args.resolution and args.filtering:
+            computed_contrasts.append(compute_contrasts_maps(curr_map,
```

Comment:
computed_contrasts can be run outside of the if with:

`computed_contrasts.append(compute_contrasts_maps(curr_map, filtering=args.filtering))`

Reply:
Then you only need to create the out_filename for each if and at the end you run the `nib.save`